worker/uniter/hook/hook.go

```diff
@@ -41,6 +41,10 @@ type Info struct {
 
 	// StorageId is the ID of the storage instance relevant to the hook.
 	StorageId string `yaml:"storage-id,omitempty"`
+
+	// DepartingUnit is the name of the unit that goes away. It is only set
+	// when Kind indicates a relation-departed hook.
+	DepartingUnit string `yaml:"departee,omitempty"`
```

Comment:
I wonder if we also want to optionally set DepartingApplication if we know that all units of the application will be going away. (a departing unit needs to remove themselves from the application config, and rebalance the cassandra cluster, etc. but a departing application just shuts it all down.)

Reply:
IIRC this was raised in a standup and we punted it for now. While we can easily get this information for the local app we cannot get it for the remote one due to lack of permissions.